impl/src/rendering/templates/partials/crates_macro.template

```diff
@@ -0,0 +1,183 @@
+# A mapping of package names to a set of normal dependencies for the Rust targets of that package.
```

Comment:
What kind of stability guarantee are we making for these maps? What about the functions? Their return types? Their argument sets?

---

I don't have strong opinions about any of these (except a general bias toward "stable" and "thought-out").

Reply:
I don't think I quite understand. The functions have a full docstring explaining what they're expected to produce with explicit failure modes (some one or two which might be too aggressive). These maps should always be present in this file and I think bucketing the cargo dependencies into bazel packages works quite nicely since the definition of both a cargo and bazel pacakge are directory based (more or less in the case of cargo but definitely works for this case).

Reply:
You're asserting here that we will have these values in this file indefinitely, but that's clearly not true. In the other comment you weren't sure whether or not users would even need the map you're providing.

Values that are being provided here are essentially part of our public api and will basically need to be preserved indefinitely (or, a migration approach provided if we find the need to change this).

The question remains -- what type of stability guarantee are we providing about the build macros here, about their arguments, and about their return types. If you're not sure, consider annotating these "experimental", so that we can break them later with impunity.